PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+            self.offeringsManager.updateOfferingsCache(appUserID: self.appUserID,
+                                                       isAppBackgrounded: isAppBackgrounded,
+                                                       completion: nil)
+        }
+    }
+
+    private func updateAllCaches(completion: ReceivePurchaserInfoBlock?) {
+        systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+            self.purchaserInfoManager.fetchAndCachePurchaserInfo(appUserID: self.appUserID,
+                                                                 isAppBackgrounded: isAppBackgrounded,
+                                                                 completion: completion)
+            self.offeringsManager.updateOfferingsCache(appUserID: self.appUserID,
+                                                       isAppBackgrounded: isAppBackgrounded,
+                                                       completion: nil)
+        }
+    }
+
+}
+
+extension Purchases: PurchaserInfoManagerDelegate {
+
+    public func purchaserInfoManagerDidReceiveUpdated(purchaserInfo: PurchaserInfo) {
+        delegate?.purchases?(self, didReceiveUpdated: purchaserInfo)
+    }
+
+}
+
+extension Purchases: PurchasesOrchestratorDelegate {
+
+    /**
+     * Called when a user initiates a promotional in-app purchase from the App Store.
+     * If your app is able to handle a purchase at the current time, run the deferment block in this method.
+     * If the app is not in a state to make a purchase: cache the defermentBlock, then call the defermentBlock
+     * when the app is ready to make the promotional purchase.
+     * If the purchase should never be made, you don't need to ever call the defermentBlock and `Purchases`
+     * will not proceed with promotional purchases.
+     *
+     * @param product `SKProduct` the product that was selected from the app store
+     */
+    @objc
+    public func shouldPurchasePromoProduct(_ product: SKProduct,
+                                           defermentBlock: @escaping DeferredPromotionalPurchaseBlock) {
+        guard let delegate = delegate else {
+            return
+        }
+
+        delegate.purchases?(self, shouldPurchasePromoProduct: product, defermentBlock: defermentBlock)
+    }
+
+}
+
+// MARK: SubscriberAttributesManager Setters.
+extension Purchases {
+
+    /**
+     * Subscriber attributes are useful for storing additional, structured information on a user.
+     * Since attributes are writable using a public key they should not be used for
+     * managing secure or sensitive information such as subscription status, coins, etc.
+     *
+     * Key names starting with "$" are reserved names used by RevenueCat. For a full list of key
+     * restrictions refer to our guide: https://docs.revenuecat.com/docs/subscriber-attributes
+     *
+     * @param attributes Map of attributes by key. Set the value as an empty string to delete an attribute.
```

Comment:
I think the format for these params in docs is actually different for Swift: 
> For Objective-C documentation the same keywords are supported, but note that the format is slightly different. In Swift you would write - returns:, but in Objective-C you write @return. See Apple's HeaderDoc User Guide for more details. Note: jazzy currently does not support all Objective-C keywords listed in this document, only @param, @return, @warning, @see, and @note.

[From Jazzy's docs](https://github.com/realm/jazzy)

Jazzy added support for DocC format, which has the added bonus of looking a lot nicer in Xcode for developers looking symbols up through the IDE, so perhaps we should try to use the DocC format for all things:
Here's the [reference](https://developer.apple.com/documentation/xcode/writing-symbol-documentation-in-your-source-files), for this it'd look like: 
```suggestion
     * - parameter attributes: Map of attributes by key. Set the value as an empty string to delete an attribute.
```

Reply:
Ugh. So. Many. Doc Strings.

Reply:
yeah, I feel you. Although I think we might be able to just find / replace the entire project in one pass

Reply:
`- Parameter` `- Warning` `- Note` too 😄

Reply:
Could have used find/replace + regex to add the `:` at the end of the param. I never remember how to regex, so I just did it by hand.